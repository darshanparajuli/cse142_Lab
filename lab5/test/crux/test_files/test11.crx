// test while-loop requires boolean condition

func main() : void {
  var i:int;
  var f:float;
  var b:bool;

  while (i) { ::printInt(7); }
  while (f) { ::printInt(8); }
  while (b) { ::printInt(9); }
}
