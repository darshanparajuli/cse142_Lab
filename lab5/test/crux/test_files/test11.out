Error type-checking file.
TypeError(8,3)[WhileLoop requires bool condition not int.]
TypeError(9,3)[WhileLoop requires bool condition not float.]

